### DISCUSSION TITLE
Train the openvino model of patchcore

### DISCUSSION THREAD

**Discussion by ShawnGao123** [open-edge-platform/anomalib#602] [Q&A] [answered]
When I trained patchcore model, I want to get the openvino model, so in the config.yaml of patchcore,  I set 

optimization:
  export_mode: openvino

But at the end of training, there is always the RuntimeError: 
Exporting the operator cdist to ONNX opset version 11 is not supported. 
Please feel free to request support or submit a pull request on PyTorch GitHub.

How can I solve this problem?

Thanks very much.

**Comment by ashwinvaidya17** [comment, marked as answer]
@ShawnGao123 This issue (https://github.com/openvinotoolkit/anomalib/issues/440) might address your problem. The exact fix depends on your pytorch version

**Reply by ShawnGao123** [reply]
Thanks! I update my torch to the version of 1.12.0, this problem is solved. The openvino model of patchcore can be trained. But when I try to  predict images with the trained model on a device without GPU, there is an error:

executable_network = ie_core.load_network(network=network, device_name="CPU")
File "ie_api.pyx", line 413, in openvino.inference_engine.ie_api.IECore.load_network
File "ie_api.pyx", line 457, in openvino.inference_engine.ie_api.IECore.load_network
RuntimeError: bad allocation

If I predict this trained model on a device with GPU, there is no the error.

Thanks!

**Reply by ilyachur** [reply]
Hi @ShawnGao123
Do you know the CPU model which you use in both cases?

Also the error happens only on your model, or it is model independent issue?

**Reply by ShawnGao123** [reply]
I have trained the openvino model of patchcore on a device with GPU, I succesfully got the model and use it to predict some images on the device with GPU.

However, I also want to use the same model to predict some images on a device without GPU, because GPU is expensive. But, it failed and there is the error:

executable_network = ie_core.load_network(network=network, device_name="CPU")
File "ie_api.pyx", line 413, in openvino.inference_engine.ie_api.IECore.load_network
File "ie_api.pyx", line 457, in openvino.inference_engine.ie_api.IECore.load_network
RuntimeError: bad allocation

Do you know how to solve this problem?

Thanks

**Reply by ilyachur** [reply]
@ShawnGao123 Do you know the CPU model which you use? I am asking this question in order to understand OpenVINO supports your CPU or not.
Also did you try to inference other models on your machine?

**Reply by ShawnGao123** [reply]
You mean the CPU model of my device? It is Intel(R) Core(TM) i7-10710U CPU @1.10GH

I use the openvino model of padim to predict some images on the my machine with above CPU (without GPU), it is successful.

But I also want to inference the  openvino model of patchcore on the same machine, because patchcore have better performance than padim.

Thanks.

**Reply by ilyachur** [reply]
@ShawnGao123 Does machine with GPU have another CPU model? Or are CPUs the same?
@dmitry-gorokhov Do you have thoughts?

**Reply by ShawnGao123** [reply]
Yes, the machine with GPU has the CPU Intel(R) W-2140B CPU@3.20 GHz

**Reply by dmitry-gorokhov** [reply]
@ShawnGao123 In most cases such error means there is no enough memory (DDR) to infer this model. What DDR capacity you have on Intel(R) Core(TM) i7-10710U? If Intel(R) W-2140B has bigger amount of memory could you please try to run OpenVINO on this CPU?

**Reply by ShawnGao123** [reply]
You are right. The machine with GPU also have a better CPU DDR capacity than the machin only with CPU.
